### PR TITLE
fix(FEC-10427): remove cues for native external subtitles 

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Issue: safari keeps the cues from external text tracks that were active and keeps playing it on the next media.
Solution: remove cues from next media won't use the text tracks that was active.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -574,7 +574,8 @@ export default class Player extends FakeEventTarget {
     if (this._reset) return;
     this.pause();
     //make sure all services are reset before engine and engine attributes are reset
-    this._externalCaptionsHandler.reset();
+    // $FlowFixMe
+    this._externalCaptionsHandler.reset(this._tracks.filter(track => track.external));
```
Comment (OrenMe) [inline src/player.js:578]: what is the issue with the typing?

File: src/track/external-captions-handler.js [hunk 2]
```diff
@@ -456,6 +458,23 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
+   * removing cues from an existing text element in a video tag
+   * @param {Array<TextTrack>} externalTracks - external tracks
+   * @return {void}
+   */
+  _removeCuesFromExternalNativeTextTrack(externalTracks: Array<TextTrack>): void {
+    const videoElement = this._player.getVideoElement();
+    if (videoElement) {
+      Array.from(externalTracks).forEach(externalTrack => {
+        const track = Array.from(videoElement.textTracks).find(track => (track ? track.language === externalTrack.language : false));
+        if (track.cues) {
```
Comment (yairans) [inline src/track/external-captions-handler.js:471]: check `track` is not null

File: src/track/external-captions-handler.js [hunk 3]
```diff
@@ -456,6 +458,23 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
+  /**
+   * removing cues from an existing text element in a video tag
+   * @param {Array<TextTrack>} externalTracks - external tracks
+   * @return {void}
+   */
+  _removeCuesFromExternalNativeTextTrack(externalTracks: Array<TextTrack>): void {
+    const videoElement = this._player.getVideoElement();
+    if (videoElement) {
+      Array.from(externalTracks).forEach(externalTrack => {
+        const track = Array.from(videoElement.textTracks).find(track => (track ? track.language === externalTrack.language : false));
```
Comment (yairans) [inline src/track/external-captions-handler.js:470]:
```suggestion
        const track = Array.from(videoElement.textTracks).find(track => (track && track.language === externalTrack.language));
```

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/player.js (hunk 1 above)
Comment (Yuvalke) [reply src/player.js:578]: the external property doesn't exist on Track class sp flow doesn't like it but filter using `undefined` same as `false` so there is no problem.

File: src/track/external-captions-handler.js [hunk 4]
```diff
@@ -456,6 +460,23 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
 
+  /**
+   * set external prefix on reset to avoid usage of the text track on the next media
+   * @param {Array<TextTrack>} externalTracks - external tracks
+   * @return {void}
+   */
+  _setPrefixForExternalNativeTextTrack(externalTracks: Array<TextTrack>): void {
+    const videoElement = this._player.getVideoElement();
+    if (videoElement) {
+      Array.from(externalTracks).forEach(externalTrack => {
```
Comment (yairans) [inline src/track/external-captions-handler.js:471]: seems like `Array.from` should be before. this method already gets `Array`

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/track/external-captions-handler.js [hunk 5]
```diff
@@ -240,7 +240,7 @@ class ExternalCaptionsHandler extends FakeEventTarget {
   reset(externalTracks: Array<TextTrack>): void {
     this._resetCurrentTrack();
     this._textTrackModel = {};
-    this._setPrefixForExternalNativeTextTrack(externalTracks);
+    this._deleteExternalNativeTextTrack(externalTracks);
```
Comment (OrenMe) [inline src/track/external-captions-handler.js:243]: delete is not the best name I think as we are not deleting it not does the HTML5 spec exposes a delete method.
```suggestion
    this._resetExternalNativeTextTrack(externalTracks);
```

**Review by OrenMe — CHANGES_REQUESTED** [review]

**Pull request merged**